server/src/main/java/com/google/fhir/proxy/HttpFhirClient.java

```diff
@@ -43,7 +43,38 @@ public abstract class HttpFhirClient {
 
   // The list of header names to keep in a response sent from the proxy; use lower case only.
   // Note we don't copy content-length/type because we may modify the response.
-  static final Set<String> HEADERS_TO_KEEP = Sets.newHashSet("last-modified", "date");
+  static final Set<String> RESPONSE_HEADERS_TO_KEEP =
+      Sets.newHashSet(
+          "last-modified",
+          "date",
+          "expires",
+          "content-location",
+          "etag",
+          "location",
+          "x-progress",
+          "x-request-id",
+          "x-correlation-id");
+
+  // The list of incoming header names to keep for forwarding request to the FHIR server; use lower
+  // case only.
+  // We should NOT copy Content-Length as this is automatically set by the RequestBuilder when
+  // setting content Entity; otherwise we will get a ClientProtocolException.
+  static final Set<String> REQUEST_HEADERS_TO_KEEP =
+      Sets.newHashSet(
+          "content-type",
+          "last-modified",
+          "etag",
+          "prefer",
+          "fhirVersion",
+          "if-match",
+          "if-none-match",
+          "if-modified-since",
+          "if-unmodified-since",
+          "if-range",
+          "x-request-id",
+          "x-correlation-id",
+          "x-forwarded-for",
+          "x-forwarded-host");
```

Comment:
Question: What is the benefit of passing this? Have you tried to see if this makes any difference on how HAPI or GCP FHIR servers respond to queries? (ditto for `x-forwarded-for`)
What I am wondering is (i) what is the scenario where we receive these and whether we should keep them and (ii) if they make a difference, maybe we should _set_ them when querying the FHIR servers (for example I am thinking if setting those may solve [this problem](https://github.com/google/fhir-access-proxy/blob/cd1e01e502b3aa4798f29f7e72f5909479417601/server/src/main/java/com/google/fhir/proxy/BearerAuthorizationInterceptor.java#L345)).